src/Plugin/QueueWorker/EloquaAppCloudQueueWorkerBase.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Drupal\eloqua_app_cloud\Plugin\QueueWorker;
+
+use Drupal\Core\Queue\QueueWorkerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+use Drupal\eloqua_app_cloud\Exception\EloquaAppCloudQueueException;
+use Psr\Log\LoggerInterface;
+
+
+abstract class EloquaAppCloudQueueWorkerBase extends QueueWorkerBase implements QueueWorkerInterface, ContainerFactoryPluginInterface {
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($queueItem) {
+
+  }
+
+  /**
+   * Wraps try/catch and retry attempts around Elomentary's $bulkApi->upload()
+   * function.
+   *
+   * @param {Object} $bulkApi
+   *    Elomentary bulkApi object reference.
+   *
+   * @param {Array} $contactsToBeStaged
+   *    Reference to the $contacts to be staged array.
+   *
+   * @param {Logger} $logger
+   *    Logger object for logging routine info / errors.
+   */
+  protected function tryBulkApiUpload($bulkApi, $contactsToBeStaged, LoggerInterface $logger) {
+    $retries = 1;
+    $maxRetries = 10;
+
+    try {
+      // Upload content to Eloqua staging area.
+      $bulkApi->upload($contactsToBeStaged);
+    } catch (Exception $e) {
+
+      $msg = "Upload Failed! Caught Exception: " . $e->getMessage();
+      $logger->error($msg);
+
+      // Retry for $maxRetries attempts.
+      for ($retries; $retries <= $maxRetries; $retries++) {
+        try {
+
+          // Upload content to Eloqua staging area.
+          $bulkApi->upload($contactsToBeStaged);
+          // Uploaded contacts successfully.
+          return;
+        } catch (Exception $e) {
```

Comment:
Should be `\Exception`?

Reply:
Good catch, copy-pasta. :(
I added a use Exception; since it is all over the place in that class.